packages/backend/src/registries/ConversationRegistry.ts

```diff
@@ -107,55 +110,72 @@ export class ConversationRegistry extends Publisher<Conversation[]> implements D
     if (messageIndex === -1)
       throw new Error(`message with id ${messageId} does not exist in conversation ${conversationId}.`);
 
-    const content = ((conversation.messages[messageIndex] as PendingChat)?.choices || [])
-      .map(choice => choice.content)
-      .join('');
-
     this.update(conversationId, messageId, {
       ...conversation.messages[messageIndex],
       choices: undefined,
       role: 'assistant',
       completed: Date.now(),
-      content: content,
     } as AssistantChat);
   }
 
   /**
-   * Utility method to quickly add a usage to a given a message inside a conversation
+   * Utility method to quickly add a usage to a conversation
    * @param conversationId
-   * @param messageId
    * @param usage
    */
-  setUsage(conversationId: string, messageId: string, usage: ModelUsage | undefined | null): void {
+  setUsage(conversationId: string, usage: ModelUsage): void {
     const conversation: Conversation = this.get(conversationId);
-
-    const messageIndex = conversation.messages.findIndex(message => message.id === messageId);
-    if (messageIndex === -1)
-      throw new Error(`message with id ${messageId} does not exist in conversation ${conversationId}.`);
-
-    this.update(conversationId, messageId, {
-      ...conversation.messages[messageIndex],
-      usage: usage,
-    } as PendingChat);
+    if (!usage) {
+      return;
+    }
```

Comment:
if usage cannot be undefined, the check from line 128 to 130 is not needed anymore